api/jobs_test.go

```diff
@@ -41,7 +41,18 @@ var ctx = context.Background()
 func TestCreateJobHandler(t *testing.T) {
 	t.Parallel()
 
-	jobsCollectionMock := &apiMock.JobStorerMock{
+	cfg, err := config.Get()
```

Comment:
So these are repetitive blocks of code (lines 44 to 54) in each test, this could be moved into a separate function that each Test calls before running any assertions. Also you are overwriting the t variable name which is set to *testing.T, better to use a different variable in for loop (line 52).

But all of the above wont matter as I don't like the idea of using the configuration values to determine whether the code in create job handler is correct, if we remove a taskName from our default configuration for validTaskNames this test will fail; but there is nothing wrong with the code.

It would be better to create a global variable map[string]**bool** and populate it with values you control in this test. Like so:

```
var taskName1, taskName2 = "dataset-api", "zebedee" // This can be set to anything and added to the map below

var taskNames = map[string]bool{
	taskName1: true,
	taskName2: true,
}
```

The reason to use bool over int, is to be able to collapse check of isPresent value in function `ParseTaskName`

Once using bool values instead of int:

change from
```
func ParseTaskName(taskNameStr string, taskNameValues map[string]int) error {
	_, isPresent := taskNameValues[taskNameStr]
	if isPresent == false {
		return apierrors.ErrTaskInvalidName
	}

	return nil
}
```
to 
```
// ParseTaskName returns ErrTaskInvalidName if it cannot find a valid TaskName for the given string representation
func ParseTaskName(taskName string, taskNames map[string]bool) error {
	if isPresent := taskNames[taskName]; !isPresent {
		return apierrors.ErrTaskInvalidName
	}

	return nil
}
```

Reply:
I had misunderstood what an enum is in go - thanks for explaining. I've made those changes now.